tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/EtcTypeChecker.scala

```diff
@@ -194,7 +203,10 @@ class EtcTypeChecker(varPool: TypeVarPool, inferPolytypes: Boolean = true) exten
             .setOnTypeFound(tt => onTypeFound(name.sourceRef, tt))
           solver.addConstraint(clause)
           // delegate the rest to the application-by-type
-          computeRec(ctx, solver, mkApp(ex.sourceRef, Seq(instantiatedType), args: _*))
+          val instantiatedExpr = mkApp(ex.sourceRef, Seq(instantiatedType), args: _*)
+          instantiatedExpr.typeErrorExplanation = ex.typeErrorExplanation
```

Comment:
In another PR, we should probably refactor this `mk*` builders to propagate explanation like this. I did this single manual propagation here, but it won't escalate well with all other expressions and their composition.